libs/core/tests/unit_tests/runnables/test_runnable_events_v2.py

```diff
@@ -1876,3 +1877,33 @@ async def generator(inputs: AsyncIterator[str]) -> AsyncIterator[str]:
             "tags": [],
         },
     ]
+
+
+async def test_with_explicit_config() -> None:
+    """Test astream events with explicit callbacks being passed."""
+    from langchain_core.prompts import ChatPromptTemplate
+
+    infinite_cycle = cycle([AIMessage(content="hello world", id="ai3")])
+    model = GenericFakeChatModel(messages=infinite_cycle)
+
+    @tool
+    async def say_hello(query: str, callbacks):
+        """Use this tool to look up which items are in the given place."""
+        template = ChatPromptTemplate.from_messages(
+            [("human", "what items are in {place}")]
+        )
+
+        chain = template | model.with_config(
```

Comment:
interestingly, requires a chain to trigger, likely due to context propagation logic going to check